File: backend/src/main/java/com/spring/electronics/category/CategoryDto.java
package com.spring.electronics.category;

import com.spring.electronics.product.ProductDto;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Getter
@Setter
@Builder
public class CategoryDto {

    private String code;

    private String name;

    private String description;

    private Set<String> superCategories;

    private Set<String> subCategories;

    private Set<ProductDto> products;

}

File: frontend/src/app/services/models/category-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ProductDto } from '../models/product-dto';
export interface CategoryDto {
  code?: string;
  description?: string;
  name?: string;
  products?: Array<ProductDto>;
  subCategories?: Array<string>;
  superCategories?: Array<string>;
}
